datacube/metadata/_eo3converter.py

```diff
@@ -0,0 +1,356 @@
+# This file is part of the Open Data Cube, see https://opendatacube.org for more information
+#
+# Copyright (c) 2015-2025 ODC Contributors
+# SPDX-License-Identifier: Apache-2.0
+"""
+STAC -> EO3 utilities.
+
+Utilities for translating STAC Items to EO3 Datasets.
+"""
+
+import dataclasses
+import uuid
+from collections.abc import Iterable, Iterator, Sequence
+from functools import singledispatch
+from typing import Any
+
+import pystac.item
+
+from datacube.index.eo3 import prep_eo3
+
+try:
+    from datacube.index.index import default_metadata_type_docs
+except ImportError:
+    from datacube.index.abstract import default_metadata_type_docs
+
+from odc.geo import CRS
+from odc.geo.geobox import GeoBox
+from odc.stac._mdtools import (
+    EPSG4326,
+    ConversionConfig,
+    _collection_id,
+    extract_collection_metadata,
+    mk_1x1_geobox,
+    mk_sample_item,
+    parse_item,
+)
+from odc.stac.model import (
+    BandKey,
+    ParsedItem,
+    RasterBandMetadata,
+    RasterCollectionMetadata,
+)
+from toolz import dicttoolz
+
+from datacube.model import Dataset, Product, metadata_from_doc
+
+# uuid.uuid5(uuid.NAMESPACE_URL, "https://stacspec.org")
+UUID_NAMESPACE_STAC = uuid.UUID("55d26088-a6d0-5c77-bf9a-3a7f3c6a6dab")
+
+# Mapping between EO3 field names and STAC properties object field names
+# EO3 metadata was defined before STAC 1.0, so we used some extensions
+# that are now part of the standard instead
+STAC_TO_EO3_RENAMES = {
+    "end_datetime": "dtr:end_datetime",
+    "start_datetime": "dtr:start_datetime",
+    "gsd": "eo:gsd",
+    "instruments": "eo:instrument",
+    "platform": "eo:platform",
+    "constellation": "eo:constellation",
+    "view:off_nadir": "eo:off_nadir",
+    "view:azimuth": "eo:azimuth",
+    "view:sun_azimuth": "eo:sun_azimuth",
+    "view:sun_elevation": "eo:sun_elevation",
+}
+
+(_eo3,) = (
```

Comment:
Is this an expensive computation?

Reply:
No, just parsing yaml documents. This is fine for now, but will almost certainly be revisited as we start pulling more functionality in from other repos.